py/worker/binary/optimizer_worker.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Generic worker which propose to worklist_scheduler optimizer."""
+
+import asyncio
+from typing import Tuple, Dict
+import warnings
+
+import os
+from absl import app
+from absl import flags
+from sight import sight
+from sight.sight import Sight
+from sight.widgets.decision import proposal
+from sight.widgets.decision.optimizers.bayes_opt_client import BayesOptOptimizerClient
+
+
+def warn(*args, **kwargs):
+  pass
+
+
+warnings.warn = warn
+
+FLAGS = flags.FLAGS
+
+
+def get_question_label_to_propose_actions():
+  return "Fvs"
+
+
+def get_question_label():
+  return "Generic"
+
+
+# def main(sight: Sight, action: Dict[str, int], is_last_action: bool) -> Tuple[float, Dict[str, int]]:
+
+#   # using actions we received from optimizer to propose actions to
+#   # worklist_scheduler
+#   outcome = asyncio.run(
+#       proposal.propose_actions(sight,
+#                                get_question_label_to_propose_actions(),
+#                                action_dict=action,
+#                                is_last_action=is_last_action))
+
+#   vals = list(action.values())
+#   # some mechanchism to calculate reward from the response of WS worker
+#   reward = sum(xi**2 for xi in vals)
+
+#   return reward, outcome
+
+
+async def propose_parallely(sight: Sight, opt_obj):
+  num_questions = int(os.getenv('NUM_QUESTIONS', '1'))
```

Comment:
this driver will be called when another client or worker proposes an optimization of another action worker.

You'll need to call decision_point here, which will provide 
- the number of questions to ask 
- the label to propose on (instead of hardwiring it via get_question_label_to_propose_actions())
- the reward python function, which you'll eval() when you need to compute it